cv-data-tasks/src/main/java/com/trihydro/tasks/actions/CleanupStaleActiveTimHoldingRecords.java

```diff
@@ -0,0 +1,174 @@
+        this.activeTimHoldingService = activeTimHoldingService;
+        this.activeTimService = activeTimService;
+    }
+
+    /**
+     * The run method is executed periodically to clean up stale ActiveTimHolding records.
+     * It performs the following steps:
+     * 1. Retrieves all active_tim_holding records from the database.
+     * 2. Identifies likely stale active_tim_holding records.
+     * 3. Checks for active_tim records with the same client_id and deletes them.
+     * 4. Deletes likely stale active_tim_holding records.
+     * 5. Clears the staleRecords set.
+     * 6. Adds remaining active_tim_holding records to the staleRecords set.
+     */
+    @Override
+    public void run() {
+        log.info("Running...");
+
+        // Retrieve all active_tim_holding records
+        List<ActiveTimHolding> currentRecords = retrieveAllActiveTimHoldingRecords();
+        log.info("Retrieved {} active_tim_holding records", currentRecords.size());
+
+        // If no stale records identified last run, this could be the first time the task runs, so add all active_tim_holding records to staleRecords set
+        if (staleRecordsIdentifiedLastRun.isEmpty()) {
+            log.info(
+                "No stale records identified last run. Adding {} active_tim_holding records to staleRecords set for next run",
+                currentRecords.size());
+            for (ActiveTimHolding record : currentRecords) {
+                staleRecordsIdentifiedLastRun.add(record.getActiveTimHoldingId());
+            }
+            return;
+        }
+
+        // Separate likely stale active_tim_holding records
+        List<ActiveTimHolding> likelyStaleRecords = new ArrayList<>();
+        List<ActiveTimHolding> newRecords = new ArrayList<>();
+        for (ActiveTimHolding record : currentRecords) {
+            if (staleRecordsIdentifiedLastRun.contains(record.getActiveTimHoldingId())) {
+                likelyStaleRecords.add(record);
+            } else {
+                newRecords.add(record);
+            }
+        }
+        log.info("Identified {} likely stale active_tim_holding records",
+            likelyStaleRecords.size());
+
+        // Check for active_tim records
+        List<ActiveTim> activeTims = retrieveAllActiveTimRecords();
+        Set<Long> deletedActiveTimIds = new HashSet<>();
+        for (ActiveTimHolding record : likelyStaleRecords) {
+            for (ActiveTim activeTim : activeTims) {
+                if (record.getClientId().equals(activeTim.getClientId()) &&
+                    !deletedActiveTimIds.contains(activeTim.getActiveTimId())) {
+                    removeActiveTimRecord(activeTim); // active_tim record is no longer up-to-date
+                    deletedActiveTimIds.add(activeTim.getActiveTimId());
+                    // TODO: Delete only if the failed update was to expire the active TIM.
+                    // TODO: Consider re-submitting the active TIM if the failed update was not meant to expire it.
```

Comment:
Note: These TODOs were added as reminders for potential task improvements to ensure they are not overlooked.